pkg/nsx/services/ipblocksinfo/ipblocksinfo.go

```diff
@@ -208,36 +208,37 @@ func (s *IPBlocksInfoService) getIPBlockCIDRsByVPCConfig(vpcConfigList []v1alpha
 		return externalIPCIDRs, privateTGWIPCIDRs, fmt.Errorf("default project not found, try later")
 	}
 
-	// for all VPC path, get VPCConnectivityProfile from VPC
-	vpcStore := &VPCStore{ResourceStore: common.ResourceStore{
+	// for all VPC path, get VPCConnectivityProfile from VPC attachment
+	vpcAttachmentStore := &VpcAttachmentStore{ResourceStore: common.ResourceStore{
 		Indexer:     cache.NewIndexer(keyFunc, cache.Indexers{}),
-		BindingType: model.VpcBindingType(),
+		BindingType: model.VpcAttachmentBindingType(),
 	}}
-	queryParam := fmt.Sprintf("%s:%s", common.ResourceType, common.ResourceTypeVpc)
-	count, err := s.SearchResource(common.ResourceTypeVpc, queryParam, vpcStore, nil)
+	queryParam := fmt.Sprintf("%s:%s", common.ResourceType, common.ResourceTypeVpcAttachment)
+	count, err := s.SearchResource(common.ResourceTypeVpcAttachment, queryParam, vpcAttachmentStore, nil)
 	if err != nil {
+		log.Error(err, "failed to query vpc attachment")
 		return externalIPCIDRs, privateTGWIPCIDRs, err
 	}
-	log.V(2).Info("successfully fetch all Vpc from NSX", "count", count)
+	log.V(2).Info("successfully fetch all Vpc Attachment from NSX", "count", count)
 
 	for vpcPath := range vpcs {
 		vpcResInfo, err := common.ParseVPCResourcePath(vpcPath)
 		if err != nil {
 			return externalIPCIDRs, privateTGWIPCIDRs, fmt.Errorf("invalid VPC path %s", vpcPath)
 		}
-
-		obj := vpcStore.GetByKey(vpcPath)
-		if obj == nil {
-			return externalIPCIDRs, privateTGWIPCIDRs, fmt.Errorf("failed to get VPC %s from NSX", vpcPath)
-		}
-		vpc := obj.(*model.Vpc)
-		log.V(2).Info("successfully fetch VPC", "path", vpcPath)
 		// for pre-created vpc, mark as default for those under default project
 		vpcProjectPath := fmt.Sprintf("/orgs/%s/projects/%s", vpcResInfo.OrgID, vpcResInfo.ProjectID)
+		vpcAttachments := vpcAttachmentStore.GetByVpcPath(vpcPath)
+		if len(vpcAttachments) == 0 {
+			log.Error(err, "failed to get vpc attachment", "vpcPath", vpcPath)
+			return externalIPCIDRs, privateTGWIPCIDRs, err
```

Comment:
Here `err` is nil, I think you need to generate a new err object to present that no VpcAttachment attaches on the VPC?